gslab_scons/tests/_side_effects.py

```diff
@@ -0,0 +1,274 @@
+import os
+import re
+import mock
+import requests
+import shutil
+import subprocess
+import _test_helpers as helpers
+
+
+def r_side_effect(*args, **kwargs):
+    '''
+    This side effect mocks the behaviour of a system call
+    on a machine with R set up for command-line use.
+    '''
+    # Get and parse the command passed to os.system()
+    command = args[0]
+    match   = helpers.command_match(command, 'R')
+
+    executable = match.group('executable')
+    log        = match.group('log')
+
+    if log is None:
+        # If no log path is specified, the R script's after 
+        # removing .R (if present) and adding .Rout.
+        source = match.group('source')
+        log    = '%s.Rout' % re.sub('\.R', '', source)
+
+    if executable == "R CMD BATCH" and log:
+        with open(log.strip(), 'wb') as log_file:
+            log_file.write('Test log\n')
+
+
+def python_side_effect(*args, **kwargs):
+    command = args[0]
+    match   = helpers.command_match(command, 'python')
+
+    if match.group('log'):
+        log_path = re.sub('(\s|>)', '', match.group('log'))
+        with open(log_path, 'wb') as log_file:
+            log_file.write('Test log')
+
+
+def matlab_side_effect(*args, **kwargs):
+    '''Mock os.system for Matlab commands'''
```

Comment:
No, although I think that `build_matlab()` doesn't produce an error and produces an empty log when `matlab` isn't a recognised executable.